### PR TITLE
Create a Geetest Ruby SDK

### PR DESCRIPTION
**Pull request by yuehaohao**

### CODE REVIEW

**Comment by flanker** [comment]
PR/git commit message 都需要完善. This is a public repo!

File: Gemfile [hunk 1]
```diff
@@ -5,3 +5,10 @@ gemspec
 
 gem "rake", "~> 12.0"
 gem "rspec", "~> 3.0"
```
Comment (flanker) [inline Gemfile]: should `rake/rspec` in development/test group ?

File: lib/geetest_ruby_sdk.rb [hunk 2]
```diff
@@ -1,6 +1,23 @@
+require "logger"
 require "geetest_ruby_sdk/version"
+require "geetest_ruby_sdk/account"
 
 module GeetestRubySdk
-  class Error < StandardError; end
-  # Your code goes here...
+  API_URL = "http://api.geetest.com"
+  JSON_FORMAT = "1"
+  NEW_FAILBACK = true
```
Comment (flanker) [inline lib/geetest_ruby_sdk.rb]: `NEW_FAILBACK` ?

File: lib/geetest_ruby_sdk.rb [hunk 3]
```diff
@@ -1,6 +1,23 @@
+require "logger"
 require "geetest_ruby_sdk/version"
+require "geetest_ruby_sdk/account"
 
 module GeetestRubySdk
-  class Error < StandardError; end
-  # Your code goes here...
+  API_URL = "http://api.geetest.com"
```
Comment (flanker) [inline lib/geetest_ruby_sdk.rb]: `BASE_URL`

File: lib/geetest_ruby_sdk/account.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      origin_challenge = GeetestRubySdk::RegisterRequest.get @geetest_id, @geetest_key, digest_mod, options
+      return { success: 0 } if origin_challenge.nil? || origin_challenge.empty?
+      { challenge: origin_challenge, gt: @geetest_id, success: 1 }
+    end
+
+    def valid? challenge='', validate='', seccode='', digest_mod = GeetestRubySdk::Encrypt::DEFAULT_DIGEST_MOD
+      secret = @geetest_key + 'geetest' + challenge
+      challenge_encode = GeetestRubySdk::Encrypt.encrypt_with secret, digest_mod: digest_mod, secret_key: @geetest_key
+      return false unless validate == challenge_encode
+  		GeetestRubySdk::ValidateRequest.post(challenge, validate, seccode, digest_mod) == GeetestRubySdk::Encrypt.encrypt_with(seccode, digest_mod: digest_mod, secret_key: @geetest_key)
```
Comment (flanker) [inline lib/geetest_ruby_sdk/account.rb]: code format?

Comment (flanker) [reply lib/geetest_ruby_sdk/account.rb]: rubocop?

File: tags [hunk 5]
```diff
@@ -0,0 +1,28 @@
+!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
```
Comment (flanker) [inline tags]: Should be ignored!

**Comment by yuehaohao** [comment]
1. Create a class method set up to init Geetest account with id and key. 
2. Use the instance to register a Geetest service and get a challenge back. 
3. Use javascripts provided by Geetest in vender to init a Geetest front end instance and start validation. 
4. Post challenge, validate and Geetest generate from front end to Geetest to finish the final validation. 
5. Add Rubocop for format and RSpec for testing.

File: lib/geetest_ruby_sdk/account.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    attr_accessor :geetest_id, :geetest_key
+
+    def initialize(geetest_id, geetest_key)
+      @geetest_id = geetest_id
+      @geetest_key = geetest_key
+    end
+
+    def register(digest_mod = GeetestRubySdk::Encrypt::DEFAULT_DIGEST_MOD, options = {})
+      origin_challenge = GeetestRubySdk::RegisterRequest.get @geetest_id, @geetest_key, digest_mod, options
+      return { success: 0 } if origin_challenge.nil? || origin_challenge.empty?
```
Comment (gracezhangyaofei) [inline lib/geetest_ruby_sdk/account.rb:32]: origin_challenge.nil? || origin_challenge.empty? => origin_challenge.blank?

**Pull request merged**